src/controller/CHIPDeviceController.h

```diff
@@ -367,6 +375,32 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public Rendezvous
      */
     CHIP_ERROR CloseBleConnection();
 #endif
+    /**
+     * @brief
+     *   Discover devices advertising as commissionable that match the long discriminator.
+     * @return CHIP_ERROR   The return status
+     */
+    CHIP_ERROR DiscoverCommissioningLongDiscriminator(uint16_t long_discriminator);
+
+    /**
+     * @brief
+     *   Discover all devices advertising as commissionable.
+     * @return CHIP_ERROR   The return status
+     */
+    CHIP_ERROR DiscoverAllCommissioning();
```

Comment:
This is a long running operation (possibly many seconds depending on back-end). Does it block the main loop when called?

Reply:
No - this just kicks the process - starts a listener and sends the initial request for all. The incoming packets trigger the delegate. The python does a sleep then calls the print function.